numpy/core/src/multiarray/ctors.c

```diff
@@ -1301,9 +1301,10 @@ _array_from_array_like(PyObject *op,
      * We skip bytes and unicode since they are considered scalars. Unicode
      * would fail but bytes would be incorrectly converted to a uint8 array.
      */
-    if (!PyBytes_Check(op) && !PyUnicode_Check(op)) {
+    if (PyObject_CheckBuffer(op) && !PyBytes_Check(op) && !PyUnicode_Check(op)) {
         PyObject *memoryview = PyMemoryView_FromObject(op);
         if (memoryview == NULL) {
+            /* TODO: Should probably not blanket ignore errors. */
```

Comment:
I wonder if we should create a task to look for all blanket `PyErr_Clear()` calls and guard them so that we at least don't catch MemoryError, and maybe KeyboardInterrupt?

Reply:
Done in #20596